### PR TITLE
make content types containing enhancedbinaryfile field type(s) editable in ezplatform admin ui

### PR DESCRIPTION
**Pull request by darinda**

### CODE REVIEW

File: bundle/Resources/translations/messages.de.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+field_definition:
```
Comment (emodric) [inline bundle/Resources/translations/messages.de.yml]: Can you use a single line for every translation, like `field_definition.enhancedbinaryfile.allowedTypes` ?

File: bundle/Resources/config/ez_field_templates.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+system:
```
Comment (emodric) [inline bundle/Resources/config/ez_field_templates.yml]: Please use 4 spaces for YML files.

File: bundle/Resources/views/content_fields.html.twig [hunk 3]
```diff
@@ -0,0 +1,4 @@
+{% extends "EzPublishCoreBundle::content_fields.html.twig" %}
+{% block enhancedbinaryfile_field %}
+    {{ block('ezbinaryfile_field') }}
+{% endblock %}
```
Comment (emodric) [inline bundle/Resources/views/content_fields.html.twig]: Missing new line at EOF here and in some other files.

File: bundle/Core/Persistence/Legacy/Content/FieldValue/Converter/Converter.php [hunk 4]
```diff
@@ -62,7 +62,7 @@ public function toFieldDefinition(StorageFieldDefinition $storageDef, FieldDefin
                     'FileSizeValidator' => array(
                         'maxFileSize' => (0 !== $storageDef->dataInt1
                             ? $storageDef->dataInt1
-                            : false),
+                            : null),
```
Comment (emodric) [inline bundle/Core/Persistence/Legacy/Content/FieldValue/Converter/Converter.php]: Why this change? Could be a breaking change.

**Comment by emodric** [comment]
Hi @darinda ! Thanks for your first contribution! :tada: I've added a couple of comments so take a look at them when you have time :)

File: bundle/Core/Persistence/Legacy/Content/FieldValue/Converter/Converter.php (hunk 4 above)
Comment (darinda) [reply bundle/Core/Persistence/Legacy/Content/FieldValue/Converter/Converter.php]: Hey @emodric ,

This is needed by the FieldMapper. If there is no maxFileSize stored for the field type yet it will be auto-assigned "false" which is not a numeric value, thus leading to an exception when trying to edit the content type in the backend.

image

**Comment by darinda** [comment]
Hey @emodric ,

thanks for the quick reply to my pull request. I updated the coding style according to your comments and commented on the possible breaking change.

File: bundle/Core/Persistence/Legacy/Content/FieldValue/Converter/Converter.php (hunk 4 above)
Comment (emodric) [reply bundle/Core/Persistence/Legacy/Content/FieldValue/Converter/Converter.php]: Thanks for the explanation. @MarioBlazek can you check if this change causes any regressions?

File: bundle/Resources/config/ez_field_templates.yml [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
+    default:
+      fielddefinition_edit_templates:
```
Comment (emodric) [inline bundle/Resources/config/ez_field_templates.yml]: Missed 4 whitespaces here :)

**Comment by emodric** [comment]
I've left one more comment and when my colleague checks that there are no regressions with `false` -> `null` change, we can merge :)

File: bundle/Core/Persistence/Legacy/Content/FieldValue/Converter/Converter.php (hunk 4 above)
Comment (emodric) [reply bundle/Core/Persistence/Legacy/Content/FieldValue/Converter/Converter.php]: Okay, it seems that this could indeed cause regressions:

https://github.com/netgen/NetgenEnhancedBinaryFileBundle/blob/master/bundle/Form/FieldTypeHandler/EnhancedFile.php#L71-L76.

You can change both those checks to add a check for null too. That should be enough I think.

Comment (darinda) [reply bundle/Core/Persistence/Legacy/Content/FieldValue/Converter/Converter.php]: Do i? If someone updates the bundle it will both update `Netgen\Bundle\EnhancedBinaryFileBundle\Form\FieldTypeHandler\EnhancedFile` and `Netgen\Bundle\EnhancedBinaryFileBundle\Core\Persistence\Legacy\Content\FieldValue\Converter\Converter` so there won't be an inconsistency.

the values "false" or "null" (depending on installed version) will only come from the `Converter` class, never from database. Or is there an edge case i don't see?

Comment (emodric) [reply bundle/Core/Persistence/Legacy/Content/FieldValue/Converter/Converter.php]: Yes, you are right, it doesn't come from the database, so this is okay!

**Comment by darinda** [comment]
Thanks for merging. Do you plan to release a new tag for this so i won't have to include `dev-master` in my composer.json referencing `NetgenEnhancedBinaryFileBundle` ?

**Comment by emodric** [comment]
Yes, we will hopefully release today, @MarioBlazek needs to fix the build first :)

**Comment by MarioBlazek** [comment]
@darinda @emodric new release has been published https://github.com/netgen/NetgenEnhancedBinaryFileBundle/releases/tag/v2.0.3.
